### ISSUE TITLE
TypeError: To define root models, use `pydantic.RootModel` rather than a field called '__root__'

### ISSUE THREAD

**Issue by vertyco** [OpenXbox/xbox-webapi-python#98]
We need compatibility with the latest pydantic version
```
  File "/mnt/nas/reclaimer/Projects/PrivateCogs/arktools/arktools.py", line 51, in <module>
    from xbox.webapi.api.client import XboxLiveClient
  File "/home/vert/.local/share/Red-DiscordBot/data/autto/cogs/Downloader/lib/xbox/webapi/api/client.py", line 23, in <module>
    from xbox.webapi.api.provider.presence import PresenceProvider
  File "/home/vert/.local/share/Red-DiscordBot/data/autto/cogs/Downloader/lib/xbox/webapi/api/provider/presence/__init__.py", line 7, in <module>
    from xbox.webapi.api.provider.presence.models import (
  File "/home/vert/.local/share/Red-DiscordBot/data/autto/cogs/Downloader/lib/xbox/webapi/api/provider/presence/models.py", line 27, in <module>
    class PresenceBatchResponse(CamelCaseModel):
  File "/home/vert/redenv/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 104, in __new__
    private_attributes = inspect_namespace(
                         ^^^^^^^^^^^^^^^^^^
  File "/home/vert/redenv/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 314, in inspect_namespace
    raise TypeError("To define root models, use `pydantic.RootModel` rather than a field called '__root__'")
TypeError: To define root models, use `pydantic.RootModel` rather than a field called '__root__'
```

**Comment by codilau** [comment]
Hi,  any chance of this issue getting fixed?
Did you find a solution @vertyco ?

**Comment by vertyco** [comment]
@codilau there is a good bit that will need to be changed for pydantic >2.x.x unfortunately. in addition to not being able to name atributes `__root__`, the way you overwrite the model config has changed, and the order in which aliases are generated have also changed, so the entire model structure will need to be looked at.

**Comment by pikulo-kama** [comment]
Hi, are there any updates on this one?

**Comment by tuxuser** [comment]
Fixed with v2.1.0
